### PR TITLE
Release Post

### PR DESCRIPTION
**Pull request by mikearnaldi**

### CODE REVIEW

File: content/blog/0012-effect-3.0.mdx [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+
+Effect is finally Production Ready!
+
+I am pleased to announce that after 5 years of work we are ready to release Effect 3.0, the first stable release of the core of the Effect Ecosystem.
+
+Starting with 3.0 the main package `effect` will follow semantic versioning:
+
+1) minor releases will include new features and new modules
+2) patch releases will include bug-fixes
+3) major releases will include breaking changes
```
Comment (IMax153) [inline content/blog/0012-effect-3.0.mdx:22]:
```suggestion
1) major releases will include breaking changes
2) minor releases will include new features and new modules
3) patch releases will include bug-fixes
```

File: content/blog/0012-effect-3.0.mdx [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+
+We do not expect new major release in the near future, we will only release a new major when a significant number of API improvements have been made and after a long period of feedback.
+
+## Quickstart
+
+Effect is a typescript library that works in every runtime & project, to start using it you can:
```
Comment (IMax153) [inline content/blog/0012-effect-3.0.mdx:28]:
```suggestion
Effect is a TypeScript library that works in every runtime & project, to start using it you can:
```

File: content/blog/0012-effect-3.0.mdx [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+- Addressing of Processes by Name
+- Actors and Entities
+- Scheduling of Cluster Singletons
+- Execution of Durable Business Workflows
+
+## The Company
+
+A year ago with the objective of making Effect as easy to use and as feature complete as possible we:
+- Incorporated [Effectful Technologies Inc](https://effectful.co/).
+- Raised a Seed Round of $redacted$ from $redacted$.
```
Comment (IMax153) [inline content/blog/0012-effect-3.0.mdx:321]: Should this still be redacted? If so we might just want to remove this information.

**Review by IMax153 — CHANGES_REQUESTED** [review]

File: content/blog/0012-effect-3.0.mdx (hunk 3 above)
Comment (mikearnaldi) [reply content/blog/0012-effect-3.0.mdx:321]: It's redacted while the PR is open, will be un-redacted on publish

File: content/blog/0012-effect-3.0.mdx [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+TypeScript is quickly becoming the de-facto standard for writing business applications, thanks to the evolution of JavaScript runtimes it is now very common to write 90% (or more) of your application code in TypeScript across both Front-End and Back-End.
+
+<Tweet id="1765083019973714292" />
+
+But the JavaScript (and TypeScript) we all know and love wasn't made for this, it started as a scripting language to automate a few simple UI tasks.
+
+When developing in TypeScript to write production-grade software we feel a lack of features, namely:
+1) error management is a pain, even when using TS we often have to deal with `unknown` error types that make our handling logic a guessing game.
+2) dependency injection is totally missing, which makes our code hard to test and testing becomes reliant on monkey-patching of modules.
+3) data modelling is challanging and mostly unsafe, TypeScript types don't exist at runtime and if we don't properly check the edges of our program (where data comes in and goes out) we rely on a lie, all seems good and safe but ready to explode at any point in time.
```
Comment (tim-smart) [inline content/blog/0012-effect-3.0.mdx:55]:
```suggestion
3) data modelling is challenging and mostly unsafe. TypeScript types don't exist at runtime, and if we don't properly check the edges of our program (where data comes in and goes out) we rely on a lie, all seems good and safe but ready to explode at any point in time.
```

File: content/blog/0012-effect-3.0.mdx [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+          ),
+        )
+      }
+      return { ok: false, error: "RequestFailed" }
+    }
+  }
+} 
+```
+
+By this time I challange every human being to look at the code and tell me if it even works according to spec, let alone being confident in making any change to it. Also we still haven't solved the issue with data validation, for that we might want to add a dependency to Zod (or similar) and add a validation step (with subsequent typed error).
```
Comment (tim-smart) [inline content/blog/0012-effect-3.0.mdx:214]:
```suggestion
By this time I challenge every human being to look at the code and tell me if it even works according to spec, let alone being confident in making any change to it. Also we still haven't solved the issue with data validation, for that we might want to add a dependency to Zod (or similar) and add a validation step (with subsequent typed error).
```

File: content/blog/0012-effect-3.0.mdx [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+    Effect.timeout("1 seconds"),
+    Effect.retry(retryPolicy),
+    Effect.flatMap(Schema.decodeUnknown(Todo)),
+    Effect.withSpan("getTodo", { attributes: { id } })
+  )
+```
+
+## The Future
+
+While the core `effect` library is now stable, the rest of the ecosystem such as `@effect/platform` and `@effect/schema` aren't yet, it will be our first priority to make the ecosystem libraries stable together with adding tons of documentation and examples.
```
Comment (tim-smart) [inline content/blog/0012-effect-3.0.mdx:308]:
```suggestion
While the core `effect` library is now stable, the rest of the ecosystem such as `@effect/platform` and `@effect/schema` aren't yet. It will be our first priority to make the ecosystem libraries stable, together with adding tons of documentation and examples.
```

File: content/blog/0012-effect-3.0.mdx [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+
+Following we plan to keep iterating and develop higher and higher levels of abstraction to solve challenging issues in the development of Production-Grade TypeScript, our next goal is to build up [Effect Cluster](https://github.com/Effect-TS/cluster) the first real JavaScript solution that enables:
```
Comment (tim-smart) [inline content/blog/0012-effect-3.0.mdx:310]:
```suggestion
Following that, we plan to keep iterating and develop higher and higher levels of abstraction to solve challenging issues in the development of Production-Grade TypeScript. Our next goal is to build up [Effect Cluster](https://github.com/Effect-TS/cluster), the first real JavaScript solution that enables:
```

File: content/blog/0012-effect-3.0.mdx [hunk 8]
```diff
@@ -0,0 +1,328 @@
...
+
+## Quickstart
+
+Effect is a TypeScript library that works in every runtime & project, to start using it you can:
+
+```bash
+npm install effect
+```
+
+And follow our [docs](https://effect.website/docs/quickstart).
```
Comment (tim-smart) [inline content/blog/0012-effect-3.0.mdx:34]:
```suggestion
And follow the quickstart guide in our [documentation](https://effect.website/docs/quickstart).
```

File: content/blog/0012-effect-3.0.mdx [hunk 9]
```diff
@@ -0,0 +1,328 @@
...
+authors:
+  - name: Michael Arnaldi
+    description: Chief Executive Officer
+    avatar: https://pbs.twimg.com/profile_images/1232338149340319751/Ln9nNWyk_400x400.jpg
+    twitter: https://twitter.com/MichaelArnaldi
+---
+
+## Announcement
+
+Effect is finally Production Ready!
```
Comment (fubhy) [inline content/blog/0012-effect-3.0.mdx:14]: I'd rephrase this. We've been communicating that "we are already using it in production" for at least a year. I think it's worth mentioning that this is not your average major release but one that has already undergone significant battle testing *in production*. The point of 3.0 (imho) is that we are also now commiting to API stability. Arguably, yes, this is also (an important) part of framing something as "production ready" but still ...

File: content/blog/0012-effect-3.0.mdx [hunk 10]
```diff
@@ -0,0 +1,328 @@
...
+
+I am pleased to announce that after 5 years of work we are ready to release Effect 3.0, the first stable release of the core of the Effect Ecosystem.
+
+Starting with 3.0 the main package `effect` will follow semantic versioning:
+
+1) major releases will include breaking changes
+2) minor releases will include new features and new modules
+3) patch releases will include bug-fixes
+
+We do not expect new major release in the near future, we will only release a new major when a significant number of API improvements have been made and after a long period of feedback.
```
Comment (fubhy) [inline content/blog/0012-effect-3.0.mdx:24]: Can we already hint at a more concrete timeline / release cadence?

File: content/blog/0012-effect-3.0.mdx [hunk 11]
```diff
@@ -0,0 +1,328 @@
...
+## The Problem
+
+TypeScript is quickly becoming the de-facto standard for writing business applications, thanks to the evolution of JavaScript runtimes it is now very common to write 90% (or more) of your application code in TypeScript across both Front-End and Back-End.
+
+<Tweet id="1765083019973714292" />
+
+But the JavaScript (and TypeScript) we all know and love wasn't made for this, it started as a scripting language to automate a few simple UI tasks.
+
+When developing in TypeScript to write production-grade software we feel a lack of features, namely:
+1) error management is a pain, even when using TS we often have to deal with `unknown` error types that make our handling logic a guessing game.
```
[the remark this answers is not in the document]
Comment (fubhy) [reply content/blog/0012-effect-3.0.mdx:53]: Agreed. Possibly even giving them sub-title importance.

File: content/blog/0012-effect-3.0.mdx [hunk 12]
```diff
@@ -0,0 +1,328 @@
...
+
+Effect is a TypeScript library that works in every runtime & project, to start using it you can:
+
+```bash
+npm install effect
+```
+
+And follow our [docs](https://effect.website/docs/quickstart).
+
+If you prefer instead to have a look at a fully working & effect-native app we've prepared a [demo cli app](https://github.com/IMax153/effect-openai) that you can directly open in [Gitpod](https://gitpod.io/) or locally (if you prefer), you'll need to provide an OpenAI API Key in order to integrate with the OpenAI API. The demo app allows you to train a model via embeddings from a set of files and then allows you to prompt the trained model with questions.
```
Comment (fubhy) [inline content/blog/0012-effect-3.0.mdx:36]: For what it's worth, I think this is the perfect example / demo. But we can iterate on this a bit to make it more accessible still (it should feature a "tour guide" documentation that shows you around and invites you to experiment with it). It would also be cool to further expand it with some custom metrics (not just span metrics & traces) and give the pre-configured grafana dashboard a bit more love.

The "tour guide" is imho the most important bit though so it's not just our usual "here's some code, go check it out by yourself".

File: content/blog/0012-effect-3.0.mdx [hunk 13]
```diff
@@ -0,0 +1,328 @@
...
+    }
+  }
+} 
+```
+
+By this time I challange every human being to look at the code and tell me if it even works according to spec, let alone being confident in making any change to it. Also we still haven't solved the issue with data validation, for that we might want to add a dependency to Zod (or similar) and add a validation step (with subsequent typed error).
+
+### Effect code
+
+By using Effect the above mess becomes just 25 lines of code (imports & formatting included):
```
Comment (fubhy) [inline content/blog/0012-effect-3.0.mdx:218]:
```suggestion
By using Effect the above mess becomes just 25 lines of highly declarative code (imports & formatting included):
```

File: content/blog/0012-effect-3.0.mdx [hunk 14]
```diff
@@ -0,0 +1,328 @@
...
+    Effect.timeout("1 seconds"),
+    Effect.retry(retryPolicy),
+    Effect.flatMap(Schema.decodeUnknown(Todo)),
+    Effect.withSpan("getTodo", { attributes: { id } })
+  )
+```
+
+## The Future
+
+While the core `effect` library is now stable, the rest of the ecosystem such as `@effect/platform` and `@effect/schema` aren't yet, it will be our first priority to make the ecosystem libraries stable together with adding tons of documentation and examples.
```
Comment (fubhy) [inline content/blog/0012-effect-3.0.mdx:308]:
```suggestion
While the core `effect` library is now stable, the rest of the ecosystem, such as `@effect/platform` and `@effect/schema`, aren't yet. It will be our first priority to make the ecosystem libraries stable, together with adding tons of documentation and examples.
```

File: content/blog/0012-effect-3.0.mdx [hunk 15]
```diff
@@ -0,0 +1,328 @@
+---
+title: Effect 3.0 (Release)
```
Comment (fubhy) [inline content/blog/0012-effect-3.0.mdx:2]: I'm missing a reference / nod to ZIO in the form of "Effect is based on proven principles, blablabla". As in "many of the fundamentals are scientifically sourced and built on giants like ZIO, etc." or something along these lines giving Effect a broader empiric evidence.

Comment (datner) [reply content/blog/0012-effect-3.0.mdx:2]: Note it will also gives effect an air of elitism, so discounting the statement beforehand might be beneficial
"Effect doesn't try to reinvent the wheel, many before us built amazing software based on well understood and described mathematical and computer science theories. We are proud and are very grateful to be standing on the shoulders of giants such as ZIO, etc."
or something like that haha

File: content/blog/0012-effect-3.0.mdx (hunk 9 above)
Comment (datner) [reply content/blog/0012-effect-3.0.mdx:14]: suggestion:
"Effect is now Production Stable."

File: content/blog/0012-effect-3.0.mdx (hunk 15 above)
Comment (mikearnaldi) [reply content/blog/0012-effect-3.0.mdx:2]: That would be untrue, there has been significant divergence from ZIO in terms of requests for example, and there is not a single math principle that inspire Effect, so no :)

Comment (datner) [reply content/blog/0012-effect-3.0.mdx:2]: Then extrapolate it to match the truth my man 😭

File: content/blog/0012-effect-3.0.mdx [hunk 16]
```diff
@@ -0,0 +1,350 @@
...
+## The Company
+
+A year ago with the objective of making Effect as easy to use and as feature complete as possible we:
+- Incorporated [Effectful Technologies Inc](https://effectful.co/).
+- Raised a Seed Round of $redacted$ from $redacted$. 
+- Hired an amazing Founding Team.
+
+As a VC-backed company that deals with Free Open Source Software, especially in recent times, we are aware of our responsibility of being absolutely clear about what is OSS and what isn't.
+
+In our case everything that is released under MIT license will remain MIT licensed, in fact in order to contribute we don't even require CLAs and the Effect organization is fully managed by the Community.
```
Comment (datner) [inline content/blog/0012-effect-3.0.mdx:348]: > The Effect organization is fully managed by the Community

With his honor the BDFL Mike, no mentions needed(?) 😆 
```suggestion
So to make it very clear, **everything that is released under MIT license will remain MIT licensed** and we don't require CLAs. The Effect organization is fully managed by the Community.
```

File: content/blog/0012-effect-3.0.mdx [hunk 17]
```diff
@@ -0,0 +1,350 @@
...
+
+To make money we will build amazing commercial products and services where we leverage Effect to its fullest power, and you'll have the choice of using them as you see fit.
```
Comment (datner) [inline content/blog/0012-effect-3.0.mdx:350]: This paragraph is a tad crass, maybe something phrased like
> "We will never charge for using effect or its ecosystem, nor will we sell 'premium' libs (<- idk, never came up, just wishful thinking on my part). Effectful Technologies will house and build its own effect-powered products and services that you can choose using as you see fit."

Will show more conviction on this matter.
I expect the people who read these line and need the reassurance to start-off with an overly-skeptical approach so clear, dry language might be better than an excited tone?

File: content/blog/0012-effect-3.0.mdx [hunk 18]
```diff
@@ -0,0 +1,350 @@
...
+
+I am pleased to announce that after 5 years of work and 3+ years of production usage we are ready to release Effect 3.0, the first stable release of the core of the Effect Ecosystem.
+
+Starting with 3.0 the main package `effect` will follow semantic versioning:
+
+1) major releases will include breaking changes
+2) minor releases will include new features and new modules
+3) patch releases will include bug-fixes
+
+We do not expect new major releases in the near future, we will only release a new major when a significant number of API improvements have been made and after a long period of feedback (6-12 months).
```
Comment (datner) [inline content/blog/0012-effect-3.0.mdx:24]: A major/year is a scary pace for folk who quiver at any accidental version updates and conceptualize a 'major' as "everything is now totally different, updating means upgrading everything, new syntax new api new gotchas. Living on the edge baybeeeeee". I know it's clarified before and here again, but the differentiation between an iteration and a whole new release should be touched upon

also, the 6-12 period is overloaded in this post, it has
- long period of feedbacks
- development of 'higher and higher levels of abstraction'
- effect/cluster

note it's a lot to promise

**Review by datner — COMMENTED** [review]
Mostly nitpicks, rephrasing, and grammar.

Otherwise this reads great!

File: content/blog/0012-effect-3.0.mdx (hunk 18 above)
Comment (mikearnaldi) [reply content/blog/0012-effect-3.0.mdx:24]: they relate to different projects, in 6-12 months we expect:
- effect cluster to be usable (currently in alpha)
- platform & co to be stable (they are already in use in prod) 
- we might release a new effect major with incorporated feedback

Note that pretty much every major JS project has a major every year, a major doesn't mean break the world
